bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/NewWizard.java

```diff
@@ -169,4 +175,37 @@ public boolean canFinish() {
 		return super.canFinish();
 	}
 
+	/**
+	 * A <code>RecentNewWizardsPreferenceManager</code> is used to store and fetch
+	 * the most recent new wizards used or created from the "Other" shortcut, which
+	 * is part of the menu manager with New Wizard actions, from preferences.
+	 *
+	 * @since 3.5
+	 */
+
+	public static class RecentNewWizardsPreferenceManager {
+
+		private static final String PLUGIN_ID = "org.eclipse.ui.workbench"; //$NON-NLS-1$
+		private static final String RECENT_NEW_MENU_ITEMS = "recentNewMenuItems"; //$NON-NLS-1$
+		private static final String SEPARATOR = ","; //$NON-NLS-1$
+
+		public List<String> getMenuShortcutsFromPreferences() {
+			IEclipsePreferences pref = InstanceScope.INSTANCE.getNode(PLUGIN_ID);
+			String preferences = pref.get(RECENT_NEW_MENU_ITEMS, null);
+			if (preferences != null && !preferences.trim().isEmpty()) {
+				return Arrays.asList(preferences.split(SEPARATOR));
+			}
+			return new ArrayList<>();
+		}
+
+		public void setMenuShortcutsToPreferences(Set<String> items) {
+			IEclipsePreferences pref = InstanceScope.INSTANCE.getNode(PLUGIN_ID);
+			pref.put(RECENT_NEW_MENU_ITEMS, String.join(SEPARATOR, items));
+			try {
+				pref.flush();
+			} catch (org.osgi.service.prefs.BackingStoreException e) {
+				e.printStackTrace();
```

Comment:
Please never ever use `e.printStackTrace();` in bundle code, we have **logging** API. As you are in the workbench, you could simply do `WorkbenchPlugin.log(e);`